ServerCore/Helpers/NotificationHelper.cs

```diff
@@ -0,0 +1,240 @@
+    /// </summary>
+    /// <param name="configuration">Configuration object for the site (containing SignalR info)</param>
+    /// <param name="isDev">Whether this is a local dev deployment</param>
+    private NotificationHelper(IConfiguration configuration, bool isDev)
+    {
+        if (!isDev)
+        {
+            // TODO: get SignalR info out of configuration
+        }
+        this.IsDev = isDev;
+    }
+
+    /// <summary>
+    /// Send a notification to all players in an event
+    /// </summary>
+    /// <param name="eventObj">The event</param>
+    /// <param name="title">Notification title</param>
+    /// <param name="content">Notification content</param>
+    /// <param name="linkUrl">Link for the notification if the player clicks it</param>
+    public static void SendNotification(Event eventObj, string title, string content, string linkUrl = null)
+    {
+        Singleton.SendNotification(new Notification() { EventID = eventObj.ID, Title = title, Content = content, LinkUrl = linkUrl });
+    }
+
+    /// <summary>
+    /// Send a notification to all players on a team
+    /// </summary>
+    /// <param name="team">The team</param>
+    /// <param name="title">Notification title</param>
+    /// <param name="content">Notification content</param>
+    /// <param name="linkUrl">Link for the notification if the player clicks it</param>
+    public static void SendNotification(Team team, string title, string content, string linkUrl = null)
+    {
+        Singleton.SendNotification(new Notification() { TeamID = team.ID, Title = title, Content = content, LinkUrl = linkUrl });
+    }
+
+    /// <summary>
+    /// Send a notification to an individual player
+    /// </summary>
+    /// <param name="player">The player</param>
+    /// <param name="title">Notification title</param>
+    /// <param name="content">Notification content</param>
+    /// <param name="linkUrl">Link for the notification if the player clicks it</param>
+    public static void SendNotification(PlayerInEvent player, string title, string content, string linkUrl = null)
+    {
+        Singleton.SendNotification(new Notification() { PlayerID = player.ID, Title = title, Content = content, LinkUrl = linkUrl });
+    }
+
+    /// <summary>
+    /// Notification sender
+    /// </summary>
+    /// <param name="notification">The notification to send</param>
+    private void SendNotification(Notification notification)
+    {
+        if (this.IsDev)
+        {
+            this.ReceiveNotification(notification);
+        }
+        else
+        {
+            // TODO: RPC this call via SignalR
+            this.ReceiveNotification(notification);
+        }
+    }
+
+    /// <summary>
+    /// Notification receiver, called by the sender (locally if isDev, via SignalR if deployed)
+    /// </summary>
+    /// <param name="notification">The notification received</param>
+    public void ReceiveNotification(Notification notification)
```

Comment:
To get this called via SignalR:
1. Dependency inject ```ServerMessageListener``` into this class (may require this class to be a DI singleton -- see other comment)
1. Add a callback for ```Notification``` messages to ServerMessageListener similar to https://github.com/PuzzleServer/mainpuzzleserver/blob/42c9ddbbbb531f5473368d2ec5a222855f0a837b/ServerCore/ServerMessages/ServerMessageListener.cs#L45
1. Add an event for ```Notification``` messages to ServerMessageListener similar to https://github.com/PuzzleServer/mainpuzzleserver/blob/42c9ddbbbb531f5473368d2ec5a222855f0a837b/ServerCore/ServerMessages/ServerMessageListener.cs#L113
1. Call EnsureInitializedAsync somewhere in this code before you want notifications to start arriving
1. Subscribe ``ReceiveNotification``` to your new event